### PR TITLE
Sa 2377 grant or demote admin automation

### PR DESCRIPTION
**Pull request by kmaranionjc**
## Issues
* [SA-2377](https://jumpcloud.atlassian.net/browse/SA-2377) - Grant Admin Script/ Automate Demote Admin

## What does this solve?
Automating admin access right to user and its bound system using powershell instead of JC admin portal
## How should this be tested?
Copy the script on the 2022-02-10-Grant-Admin-Rights.md file and run it locally. It will prompt for username, systemid, and amount minutes a user is granted admin access

### CODE REVIEW

File: _posts/2022-02-10-Grant-Admin-Rights.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    Set-JcSdkSystemAssociation -systemid $SystemId -id $UserId -op 'update' -type 'user' -Attributes $attributes.AdditionalProperties
+    # Sleep for x amount of minutes then sets the user back to old permissions
+    Start-Sleep -Seconds $SleepCalculated
+    # Sets the user back to old permissions
+    Set-JcSdkSystemAssociation -systemid $SystemId -id $UserId -op 'update' -type 'user'
+    Write-Warning "Please re-login if on Windows machine to update admin rights"
+}
+$user = Get-User
+$SysId = Get-SystemId -UserId $user
+$Mins = Read-Host "How many minutes would you like to elevate this user for to Sudo/Admin for?"
```
Comment (jworkmanjc) [inline _posts/2022-02-10-Grant-Admin-Rights.md:150]: "How many minutes would you like to elevate this user for to Sudo/Admin for?" could be "How many minutes should this user be elevated as Sudo/Admin?" or similar.

File: _posts/2022-02-10-Grant-Admin-Rights.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    # If no sysId then print other systems linked to the User
+    Write-Host "User with Id $UserId will be granted Admin permission on System with Id $SystemId"
+    Write-Host "They will remain admin until $Mins minutes from now..."
+    Write-Warning "Do not close this window until the script is finished or the user will remain admin!"
+    # Calculate the seconds to sleep for 
+    $SleepCalculated = $Mins * 60
+    # Admin attributes
+    $attributes = [JumpCloud.SDK.V2.Models.IGraphOperationSystemAttributes]@{ 'sudo' = @{'enabled' = $true; 'withoutPassword' = $false } }
+    # Set the user to admin
+    Set-JcSdkSystemAssociation -systemid $SystemId -id $UserId -op 'update' -type 'user' -Attributes $attributes.AdditionalProperties
```
Comment (jworkmanjc) [inline _posts/2022-02-10-Grant-Admin-Rights.md:141]: This line will likely change when we next update our SDKs, just taking a note here more for my own notes

File: _posts/2022-02-10-Grant-Admin-Rights.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+![Grant admin Jc portal](./../../../images/grantAdminJcPortal.gif)
+### Script
+
+```powershell
+function Get-User 
+{
+    $Username = Read-Host "Please enter the target JumpCloud User you'd like to elevate to admin/sudo temporarily"
+
+    # Get users
+    $users = Get-JCUser -UserName $UserName
```
Comment (jworkmanjc) [inline _posts/2022-02-10-Grant-Admin-Rights.md:43]: The other nice benefit of using Get-JCUser here is that it calls the Connect-JCOnline function and prompts the user to enter their API key
Screen Shot 2022-02-14 at 3 28 46 PM

**Review by jworkmanjc — APPROVED** [review]
Really cool workflow. Something folks have asked for for a while and easy enough to use. 

I've tested the workflow and can confirm it works as expected. The line stating that "How many minutes would you like to elevate this user for to Sudo/Admin for?" has one extra word and should be edited before merging to the main branch

**Comment by kmaranionjc** [comment]
Was wondering why it was different lol

**Pull request merged**